pkg/sdk/testint/tags_integration_test.go

```diff
@@ -281,3 +283,915 @@ func TestInt_TagsShowByID(t *testing.T) {
+			},
+		},
+		{
+			name:       "Procedure",
+			objectType: sdk.ObjectTypeProcedure,
+			setupObject: func() sdk.SchemaObjectIdentifierWithArguments {
+				// cleanup is set up in the Create procedure
+				procedure := testClientHelper().Procedure.Create(t, sdk.DataTypeInt)
+				return procedure.ID()
+			},
+			setTags: func(id sdk.SchemaObjectIdentifierWithArguments, tags []sdk.TagAssociation) error {
+				return client.Procedures.Alter(ctx, sdk.NewAlterProcedureRequest(id).WithSetTags(tags))
+			},
+			unsetTags: func(id sdk.SchemaObjectIdentifierWithArguments, tags []sdk.ObjectIdentifier) error {
+				return client.Procedures.Alter(ctx, sdk.NewAlterProcedureRequest(id).WithUnsetTags(tags))
+			},
+		},
+	}
+
+	for _, tc := range schemaObjectWithArgumentsTestCases {
+		t.Run(fmt.Sprintf("schema object with arguments %s", tc.name), func(t *testing.T) {
+			id := tc.setupObject()
+			err := tc.setTags(id, tags)
+			require.NoError(t, err)
+
+			returnedTagValue, err := client.SystemFunctions.GetTag(ctx, tag.ID(), id, tc.objectType)
+			require.NoError(t, err)
+			assert.Equal(t, tagValue, returnedTagValue)
+
+			err = tc.unsetTags(id, unsetTags)
+			require.NoError(t, err)
+
+			_, err = client.SystemFunctions.GetTag(ctx, tag.ID(), id, tc.objectType)
+			require.ErrorContains(t, err, "sql: Scan error on column index 0, name \"TAG\": converting NULL to string is unsupported")
+		})
+	}
+}
+
+func createDatabaseFromShare(t *testing.T) (*sdk.Database, func()) {
+	t.Helper()
+	client := testClient(t)
+	secondaryClient := testSecondaryClient(t)
+	ctx := testContext(t)
+
+	shareTest, shareCleanup := secondaryTestClientHelper().Share.CreateShare(t)
+	t.Cleanup(shareCleanup)
+
+	sharedDatabase, sharedDatabaseCleanup := secondaryTestClientHelper().Database.CreateDatabase(t)
+	t.Cleanup(sharedDatabaseCleanup)
+
+	databaseId := sharedDatabase.ID()
+
+	err := secondaryClient.Grants.GrantPrivilegeToShare(ctx, []sdk.ObjectPrivilege{sdk.ObjectPrivilegeUsage}, &sdk.ShareGrantOn{
+		Database: sharedDatabase.ID(),
+	}, shareTest.ID())
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		err := secondaryClient.Grants.RevokePrivilegeFromShare(ctx, []sdk.ObjectPrivilege{sdk.ObjectPrivilegeUsage}, &sdk.ShareGrantOn{
+			Database: sharedDatabase.ID(),
+		}, shareTest.ID())
+		require.NoError(t, err)
+	})
+
+	err = secondaryClient.Shares.Alter(ctx, shareTest.ID(), &sdk.AlterShareOptions{
+		IfExists: sdk.Bool(true),
+		Set: &sdk.ShareSet{
+			Accounts: []sdk.AccountIdentifier{
+				testClientHelper().Account.GetAccountIdentifier(t),
+			},
+		},
+	})
+	require.NoError(t, err)
+
+	err = client.Databases.CreateShared(ctx, databaseId, shareTest.ExternalID(), &sdk.CreateSharedDatabaseOptions{})
+	require.NoError(t, err)
+
+	database, err := client.Databases.ShowByID(ctx, databaseId)
+	require.NoError(t, err)
+
+	return database, testClientHelper().Database.DropDatabaseFunc(t, database.ID())
+}
+
+func createDatabaseReplica(t *testing.T) (*sdk.Database, func()) {
```

Comment:
let's use the methods present in the client helper (CreateSecondaryDatabaseWithOptions)

Reply:
Actually, this involves a more complex setup with dependencies. Moved to a different file.